server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/AddressMappingRoutes.java

```diff
@@ -0,0 +1,106 @@
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.routes;
+
+import static spark.Spark.halt;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.mail.internet.AddressException;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.rrt.api.RecipientRewriteTable;
+import org.apache.james.rrt.api.RecipientRewriteTableException;
+import org.apache.james.rrt.lib.MappingSource;
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.eclipse.jetty.http.HttpStatus;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+public class AddressMappingRoutes implements Routes {
+
+    static final String BASE_PATH = "/mappings/address/";
+    static final String POST_PATH = "/mappings/address/:mappingSource/targets/:destinationAddress";
+
+    private final RecipientRewriteTable recipientRewriteTable;
+
+    @Inject
+    AddressMappingRoutes(RecipientRewriteTable recipientRewriteTable) {
+        this.recipientRewriteTable = recipientRewriteTable;
+    }
+
+    @Override
+    public String getBasePath() {
+        return BASE_PATH;
+    }
+
+    @Override
+    public void define(Service service) {
+        service.post(POST_PATH, this::addAddressMapping);
+    }
+
+    @POST
+    @Path(POST_PATH)
+    @ApiOperation(value = "Getting all user mappings in RecipientRewriteTable")
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "No body on created", response = List.class),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid parameter values.")
+    })
+
+    public HaltException addAddressMapping(Request request, Response response) {
+        MailAddress mappingSource = extractMailAddress(request, "mappingSource");
+        MailAddress destinationAddress = extractMailAddress(request, "destinationAddress");
+
+        try {
+             recipientRewriteTable.addAddressMapping(
+                 MappingSource.fromMailAddress(mappingSource),
+                 destinationAddress.asString());
+        } catch (RecipientRewriteTableException e) {
+            throw ErrorResponder.builder()
+                 .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
+                 .message(e.getMessage())
+                 .haltError();
+        }
+        return halt(HttpStatus.NO_CONTENT_204);
+    }
+
+    MailAddress extractMailAddress(Request request, String address) {
```

Comment:
This `extractMailAddress` is duplicates the `MailAddressParser` class. Please use MailAddressParser directly here to avoid repeating already written code.